lib/waffle/storage/google/cloud_storage.ex

```diff
@@ -78,10 +80,8 @@ defmodule Waffle.Storage.Google.CloudStorage do
   """
   @spec conn(String.t()) :: Tesla.Env.client()
   def conn(scope \\ @full_control_scope) do
-    token_store =
-      Application.get_env(:waffle, :token_fetcher, Waffle.Storage.Google.Token.DefaultFetcher)
-
-    token_store.get_token(scope)
+    scope
+    |> @token_fetcher.get_token()
```

Comment:
Any reason why not just leave it as a runtime dependency? Making it compile-time seems like it could limit people's options for not a lot of gain.


I mean, _ideally_ we'd be able to pass the fetcher module as options more easily, or it would be namespaced to gcs, or even the buckets specifically (I'm imagining people swapping from S3 to GCS or vice-versa), but that seems like a larger undertaking with ramifications for the other waffle storage libraries

Reply:
Thank you @petermueller , I made it back to be a runtime dependency